app-backend/db/src/main/scala/CampaignDao.scala

```diff
@@ -95,22 +101,24 @@ object CampaignDao extends Dao[Campaign] with ObjectPermissions[Campaign] {
   def insertCampaign(
       campaignCreate: Campaign.Create,
       user: User
-  ): ConnectionIO[Campaign] =
-    (fr"INSERT INTO" ++ tableF ++ fr"""(
+  ): ConnectionIO[Campaign] = {
+    for {
+      insertId <- (fr"INSERT INTO" ++ tableF ++ fr"""(
       id, created_at, owner, name, campaign_type, description,
       video_link, partner_name, partner_logo, parent_campaign_id,
       continent, tags, resource_link
     )""" ++
-      fr"""VALUES
+        fr"""VALUES
       (uuid_generate_v4(), now(), ${user.id}, ${campaignCreate.name},
        ${campaignCreate.campaignType}, ${campaignCreate.description},
        ${campaignCreate.videoLink}, ${campaignCreate.partnerName},
        ${campaignCreate.partnerLogo}, ${campaignCreate.parentCampaignId},
        ${campaignCreate.continent}, ${campaignCreate.tags}, ${campaignCreate.resourceLink}
        )
-    """).update.withUniqueGeneratedKeys[Campaign](
-      fieldNames: _*
-    )
+    """).update.withUniqueGeneratedKeys[UUID]("id")
```

Comment:
The returning here was a little weird with the addition of the `imageCount` in campaign. That value is dynamic and inserted on the fly with a select query. I didn't want to create a case class that only had one new field so I broke this (and another update/insert` into two queries: first the insert returns the ID and *then* we query for the campaign by id.

Reply:
Ha I actually had a comment written up asking about this -- thanks for explaining